### PR TITLE
16307 orm and oml java classes

### LINKED ISSUE #16307: Create HAPI HL7 classes for ORM and OML Messages
### User Story

As a Sender of ORM and OML versions 2.5.1 -> 2.7 greater,
I want the RS UP to losslessly process my message,
so that the receiver can receive all the data it contains

### Description/Use Case

Presently, for deserializing HL7v2 messages, RS uses the built-in definitions of the ORM and OML classes provided by the HAPI HL7v2 library. These classes are not fully compatible with all the different versions of ORM and OML RS may receive, and are incomplete in many cases. Thus, just like how we did for ORU_R01 in `prime-router/src/main/java/fhirengine/translation/hl7/structures/fhirinventory`, we want to add our own classes based on the following:
- CARISTIX ORM and OML definitions for all version 2.5.1+
- HL7v2 -> FHIR Mapping Inventory 
- LOI Nist Profile Specification

### Dev Notes

This will be very similar to what we did for ORU_R01 where the goal is to undeprecate all fields and expand the type to the largest version in the spec (i.e. some 2.5.1 fields are CE, but in 2.7 they are CWE, so the a new class should set them as CWE)

### Acceptance Criteria

- OML classes created in `prime-router/src/main/java/fhirengine/translation/hl7/structures/fhirinventory`
- ORM classes created in `prime-router/src/main/java/fhirengine/translation/hl7/structures/fhirinventory`

### PR DESCRIPTION
**Pull request by jack-h-wang**
This PR adds `OML_O21` and `ORM_O01` catchall Java classes. Some `ORU_R01` classes have also been updated where new classes were applicable.

Test Steps:
1. Ensure all test suites pass.

## Changes
- Add OML_O21 classes
- Add ORM_O01 classes
- Update ORU_R01 classes
- Remove dependency on HAPI v2.6 classes

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`?
- [x] Added tests?

## Linked Issues
- Fixes #16307

## To Be Done
This is the final PR to be merged into `OML-Updates-Base` before we begin the process of merging to `main`.

### CODE REVIEW

**Comment by jack-h-wang** [comment]
Note the `end2end_up` test data needs to be updated to allow the test to pass in this branch, but I'm electing to wait until this is merged into `OML-Updates-Base` so that we can do all needed test updates in the PR to merge into `main`.

File: prime-router/src/main/java/fhirengine/translation/hl7/structures/fhirinventory/group/ORM_O01_ORDER.java
Comment (JFisk42) [inline prime-router/src/main/java/fhirengine/translation/hl7/structures/fhirinventory/group/ORM_O01_ORDER.java:42]: Commenting here for visibility: Looks like ORM_O01_ORDER_DETAIL.java needs our `DG1` added.

File: prime-router/src/main/java/fhirengine/translation/hl7/structures/fhirinventory/group/ORM_O01_ORDER.java
Comment (JFisk42) [inline prime-router/src/main/java/fhirengine/translation/hl7/structures/fhirinventory/group/ORM_O01_ORDER.java:41]: Commenting here for visibility: Looks like OML_O21_ORDER.java needs our `FT1` added.

**Review by JFisk42 — COMMENTED** [review]
Reviewed all of the `/group/` changes, will review `/segment/` changes in the morning. Just a couple small additions and two questions.

File: prime-router/src/main/java/fhirengine/translation/hl7/structures/fhirinventory/group/OML_O21_OBSERVATION_REQUEST.java
Comment (jack-h-wang) [reply prime-router/src/main/java/fhirengine/translation/hl7/structures/fhirinventory/group/OML_O21_OBSERVATION_REQUEST.java:41]: The decisions to create a new class are based entirely on whether the HAPI 2.7 classes are fully capable of storing all fields specified in 2.5.1. To take RXO as an example, we did not create a version of the segment because the 2.7 class can fully represent a 2.5.1 message (CE is represented by CWE, and there are no NULLDT segments). The segments that we created classes for are in the cases where the HAPI 2.7 classes were deficient in some way; this can only be determined on a case by case basis.

File: prime-router/src/main/java/fhirengine/translation/hl7/structures/fhirinventory/group/OML_O21_INSURANCE.java
Comment (jack-h-wang) [reply prime-router/src/main/java/fhirengine/translation/hl7/structures/fhirinventory/group/OML_O21_INSURANCE.java]: I've mostly been trusting IntelliJ on this, but I went through and removed all the wildcard imports.

**Review by JFisk42 — APPROVED** [review]

File: prime-router/src/main/java/fhirengine/translation/hl7/structures/fhirinventory/group/OML_O21_OBSERVATION_REQUEST.java
Comment (JFisk42) [reply prime-router/src/main/java/fhirengine/translation/hl7/structures/fhirinventory/group/OML_O21_OBSERVATION_REQUEST.java:42]: Thanks for the explanation!

**Comment by JFisk42** [comment]
> Note the `end2end_up` test data needs to be updated to allow the test to pass in this branch, but I'm electing to wait until this is merged into `OML-Updates-Base` so that we can do all needed test updates in the PR to merge into `main`.

Makes sense. There will already me a "merge conflicts test updates" commit for the PR back to main so this will be easy to add.

File: prime-router/src/main/java/fhirengine/translation/hl7/structures/fhirinventory/segment/DG1.java
Comment (jack-h-wang) [reply prime-router/src/main/java/fhirengine/translation/hl7/structures/fhirinventory/segment/DG1.java]: I went ahead and marked this as not required. This was one of the ways I was checking for differences, but I must have missed this one.

**Comment by jalbinson** [comment]
Once this is merged into `OML-Updates-Base`, then you will update the e2e tests? Will there be another review from `OML-Updates-Base` -> `main`?

File: prime-router/src/main/java/fhirengine/translation/hl7/structures/fhirinventory/group/OML_O21_INSURANCE.java
Comment (jalbinson) [reply prime-router/src/main/java/fhirengine/translation/hl7/structures/fhirinventory/group/OML_O21_INSURANCE.java]: I'd say most of the time wildcard imports are not a problem but dealing with mapping classes that have duplicate class names everywhere makes this a bit more prone to errors.

**Comment by jalbinson** [comment]
Are any unit tests required for our fields that have been "widened" or are existing tests sufficient?

**Comment by jack-h-wang** [comment]
> Once this is merged into `OML-Updates-Base`, then you will update the e2e tests? Will there be another review from `OML-Updates-Base` -> `main`?

There will have to be, since `OML-Updates-Base` will have a lot of tests break once we try to rebase it on main. It's best to defer all the test updates until then.

**Review by jalbinson — COMMENTED** [review]
Unit and integration tests passed locally. Looks good to me!

**Pull request merged**
